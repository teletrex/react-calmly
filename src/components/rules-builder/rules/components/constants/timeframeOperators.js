/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import { RULE_INPUT_TYPES } from '../../../constants';

const { NONE, RANGE, SINGLE_VALUE, WHEN_RELATION, WITHIN_PERIOD } = RULE_INPUT_TYPES;

export const TIMEFRAME_OPERATORS = [
  // Within:
  {
    inputType: WITHIN_PERIOD,
    label: 'within the last',
    value: 'WITHIN_THE_LAST',
  },

  // Today:
  {
    inputType: NONE,
    label: 'on today',
    value: 'ON_TODAY',
  },
  {
    inputType: NONE,
    label: 'before today',
    value: 'BEFORE_TODAY',
  },

  // Any time:
  {
    inputType: NONE,
    label: 'at any time',
    value: 'AT_ANY_TIME',
  },

  // Date:
  {
    inputType: SINGLE_VALUE,
    label: 'on',
    value: 'ON',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'after',
    value: 'AFTER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'before',
    value: 'BEFORE',
  },

  // Date range:
  {
    inputType: RANGE,
    label: 'between',
    value: 'BETWEEN',
  },

  // When relation:
  {
    inputType: WHEN_RELATION,
    label: 'exactly',
    value: 'EXACTLY',
  },
  {
    inputType: WHEN_RELATION,
    label: 'more than',
    value: 'MORE_THAN',
  },
  {
    inputType: WHEN_RELATION,
    label: 'less than',
    value: 'LESS_THAN',
  },
];
