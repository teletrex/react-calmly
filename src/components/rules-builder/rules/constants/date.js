/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export const DATE_PATTERN = 'd{1,2}/d{1,2}/d{4}';
