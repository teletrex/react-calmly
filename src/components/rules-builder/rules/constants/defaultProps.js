/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export const DEFAULT_PROPS = {
  INPUT_FIELD_VALIDATION: [],
  OPERATORS: [],
  TIMEFRAME: {},
};
