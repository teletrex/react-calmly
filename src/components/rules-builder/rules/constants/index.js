/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export * from './date';
export * from './defaultProps';
export * from './periods';
export * from './propTypes';
export * from './ruleOperators';
export * from './validationMessages';
export * from './values';
export * from './whenRelations';
