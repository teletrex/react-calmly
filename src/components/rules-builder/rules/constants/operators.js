/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import { RULE_INPUT_TYPES } from '../../constants';

const {
  MULTIPLE_VALUES,
  NONE,
  PERIOD_NUMBER,
  PERIOD_RANGE,
  RANGE,
  SINGLE_VALUE,
  WHEN_RELATION,
  WITHIN_PERIOD,
} = RULE_INPUT_TYPES;

// Common operators
export const BLANK_OPERATORS = [
  {
    inputType: NONE,
    label: 'is blank',
    value: 'IS_BLANK',
  },
  {
    inputType: NONE,
    label: 'is not blank',
    value: 'IS_NOT_BLANK',
  },
];

export const EXACT_OPERATORS = [
  {
    inputType: WHEN_RELATION,
    label: 'is exactly',
    value: 'IS_EXACTLY',
  },
  {
    inputType: WHEN_RELATION,
    label: 'is not exactly',
    value: 'IS_NOT_EXACTLY',
  },
];

export const IS_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is',
    value: 'IS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is not',
    value: 'IS_NOT',
  },
];

export const HAS_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'has',
    value: 'HAS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'has not',
    value: 'HAS_NOT',
  },
];

export const IS_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'is one of the following',
    value: 'IS_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'is not one of the following',
    value: 'IS_NOT_ONE_OF_THE_FOLLOWING',
  },
];

export const CONTAIN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'contains',
    value: 'CONTAINS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not contain',
    value: 'DOES_NOT_CONTAIN',
  },
];

export const CONTAIN_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'contains one of the following',
    value: 'CONTAINS_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not contain one of the following',
    value: 'DOES_NOT_CONTAIN_ONE_OF_THE_FOLLOWING',
  },
];

export const RANGE_OPERATORS = [
  {
    inputType: RANGE,
    label: 'is between',
    value: 'IS_BETWEEN',
  },
];

// String related operators
export const STARTS_WITH_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'starts with',
    value: 'STARTS_WITH',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not start with',
    value: 'DOES_NOT_START_WITH',
  },
];

export const STARTS_WITH_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'starts with one of the following',
    value: 'STARTS_WITH_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not start with one of the following',
    value: 'DOES_NOT_START_WITH_ONE_OF_THE_FOLLOWING',
  },
];

export const ENDS_WITH_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'ends with',
    value: 'ENDS_WITH',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not end with',
    value: 'DOES_NOT_END_WITH',
  },
];

export const ENDS_WITH_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'ends with one of the following',
    value: 'ENDS_WITH_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not end with one of the following',
    value: 'DOES_NOT_END_WITH_ONE_OF_THE_FOLLOWING',
  },
];

// Number related operators
export const LESS_THAN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is less than',
    value: 'IS_LESS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is less than or equal to',
    value: 'IS_LESS_OR_EQUAL',
  },
];

export const GREATER_THAN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is greater than',
    value: 'IS_GREATER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is greater than or equal to',
    value: 'IS_GREATER_OR_EQUAL',
  },
];

// Date related operators
export const TODAY_OPERATORS = [
  {
    inputType: NONE,
    label: 'is today',
    value: 'IS_TODAY',
  },
  {
    inputType: NONE,
    label: 'is after today',
    value: 'IS_AFTER_TODAY',
  },
  {
    inputType: NONE,
    label: 'is not after today',
    value: 'IS_NOT_AFTER_TODAY',
  },
  {
    inputType: NONE,
    label: 'is before today',
    value: 'IS_BEFORE_TODAY',
  },
  {
    inputType: NONE,
    label: 'is not before today',
    value: 'IS_NOT_BEFORE_TODAY',
  },
  {
    inputType: NONE,
    label: 'anniversary is today',
    value: 'ANNIVERSARY_IS_TODAY',
  },
];

export const ANNIVERSARY_OPERATORS = [
  {
    inputType: WHEN_RELATION,
    label: 'anniversary is',
    value: 'ANNIVERSARY_IS',
  },
];

export const WITHIN_OPERATORS = [
  {
    inputType: WITHIN_PERIOD,
    label: 'is within the last',
    value: 'IS_WITHIN_THE_LAST',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is not within the last',
    value: 'IS_NOT_WITHIN_THE_LAST',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is within the next',
    value: 'IS_WITHIN_THE_NEXT',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is not within the next',
    value: 'IS_NOT_WITHIN_THE_NEXT',
  },
];

export const WITHIN_RANGE_OPERATORS = [
  {
    inputType: PERIOD_RANGE,
    label: 'is within the range of the last',
    value: 'IS_WITHIN_THE_RANGE_OF_THE_LAST',
  },
];

export const PERIOD_EQUAL_OPERATORS = [
  {
    inputType: PERIOD_NUMBER,
    label: 'year is',
    value: 'YEAR_IS',
  },
  {
    inputType: PERIOD_NUMBER,
    label: 'month is',
    value: 'MONTH_IS',
  },
  {
    inputType: PERIOD_NUMBER,
    label: 'day is',
    value: 'DAY_IS',
  },
];

export const BEFORE_AFTER_DATE_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is after',
    value: 'IS_AFTER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is before',
    value: 'IS_BEFORE',
  }
]

export const DATA_AGGREGATION_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'by',
    value: 'BY',
  },

];
