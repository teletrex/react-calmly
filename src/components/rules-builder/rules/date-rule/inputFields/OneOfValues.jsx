/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import React, { Fragment, useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import map from 'lodash/map';
import uniq from 'lodash/uniq';
import without from 'lodash/without';

import { DatePickerInput, Tag, DatePicker } from '@carbon/react';
import RulesBuilderContext from '../../../RulesBuilderContext';
import { DATE_PATTERN } from '../../constants';
import { getPrefixedClasses } from '../../helpers';

const parseDate = timestamp => new Date(timestamp).toDateString();

const OneOfValues = ({ id, onValueChange, value }) => {
  const { isReadOnly } = useContext(RulesBuilderContext);
  const [timestamps, setTimestamps] = useState(value);

  useEffect(() => {
    setTimestamps(value);
  }, [value]);

  const addValue = dateValue => {
    const date = new Date(dateValue[0]);
    const newTimestamps = uniq([...timestamps, date.getTime()]);

    setTimestamps(newTimestamps);
    onValueChange(newTimestamps);
  };

  const removeValue = timestamp => {
    const newTimestamps = without(timestamps, timestamp);

    setTimestamps(newTimestamps);
    onValueChange(newTimestamps);
  };

  if (isReadOnly) {
    return map(timestamps, (timestamp, i) => (
      <Fragment key={timestamp}>
        <span>{i ? ',' : ':'}</span>
        <span className={getPrefixedClasses(['--text-space-left', '--rule-value-readonly'])}>
          {parseDate(timestamp)}
        </span>
      </Fragment>
    ));
  }

  return (
    <div
      className={getPrefixedClasses('--input-field-container')}
      data-testid="date-rule-one-of-values"
    >
      {map(timestamps, timestamp => (
        <Tag
          key={timestamp}
          filter
          onClick={() => {
            removeValue(timestamp);
          }}
          title={timestamp.toString()}
        >
          {parseDate(timestamp)}
        </Tag>
      ))}
      <DatePicker datePickerType="single" id={`${id}--date-rule-picker-1`} onChange={addValue}>
        <DatePickerInput id={`${id}--date-rule-value-1`} labelText="" pattern={DATE_PATTERN} />
      </DatePicker>
    </div>
  );
};

OneOfValues.propTypes = {
  id: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
  value: PropTypes.arrayOf(PropTypes.number),
};

OneOfValues.defaultProps = {
  value: [],
};

export default OneOfValues;
