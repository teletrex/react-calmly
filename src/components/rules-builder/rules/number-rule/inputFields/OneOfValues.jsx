/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */




import React, { useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import map from 'lodash/map';
import uniq from 'lodash/uniq';
import  {useTranslation}  from '../../../../../translation';
import AddTags from '../../../../add-tags';
import RulesBuilderContext from '../../../RulesBuilderContext';
import { PROPTYPES } from '../../constants';
import { getPrefixedClasses } from '../../helpers';

const OneOfValues = ({ id, onValueChange, operator, placeholder, value }) => {
  const { t } = useTranslation();
  const { isReadOnly } = useContext(RulesBuilderContext);

  const [valueTags, setValueTags] = useState(map(value, String));
  const [inputValue, setInputValue] = useState('');

  useEffect(() => {
    setInputValue('');
  }, [operator]);

  useEffect(() => {
    setValueTags(map(value, String));
  }, [value]);

  const handleAddTag = tags => {
    const newTags = uniq(map(tags, tag => tag.trim()));

    setValueTags(newTags);
    onValueChange(map(newTags, Number));
    setInputValue('');
  };

  const handleDeleteTag = tags => {
    setValueTags(tags);
    onValueChange(map(tags, Number));
  };

  const handleInputChange = newInputValue => {
    setInputValue(newInputValue);
  };

  if (isReadOnly) {
    return (
      <span className={getPrefixedClasses(['--rule-value-readonly', '--text-space-left'])}>
        {valueTags.join(', ')}
      </span>
    );
  }

  return (
    <AddTags
      allowTagSplitting={false}
      id={id}
      inline
      inlineInputAlwaysOn
      inputValue={inputValue}
      min={0}
      name={id}
      onAddTag={handleAddTag}
      onDelete={handleDeleteTag}
      onInputChange={handleInputChange}
      placeholder={placeholder || t('Enter value')}
      shapeForm="oval"
      tagExistsError={t('This value already exists')}
      tags={valueTags}
      type="number"
      validateInputValue
    />
  );
};

OneOfValues.propTypes = {
  id: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
  operator: PROPTYPES.OPERATOR,
  placeholder: PropTypes.string,
  value: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])),
};

OneOfValues.defaultProps = {
  operator: null,
  placeholder: '',
  value: [],
};

export default OneOfValues;
