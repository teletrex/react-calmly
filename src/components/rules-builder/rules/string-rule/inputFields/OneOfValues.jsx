/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */




import React, { useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import map from 'lodash/map';
import uniq from 'lodash/uniq';

import AddTags from '../../../../add-tags';
import  {useTranslation}  from '../../../../../translation';
import RulesBuilderContext from '../../../RulesBuilderContext';
import { PROPTYPES } from '../../constants';
import { getPrefixedClasses } from '../../helpers';

const OneOfValues = ({ id, onValueChange, operator, placeholder, value }) => {
  const { t } = useTranslation();
  const { isReadOnly } = useContext(RulesBuilderContext);

  const [valueTags, setValueTags] = useState(value);
  const [inputValue, setInputValue] = useState('');

  useEffect(() => {
    setInputValue('');
  }, [operator]);

  useEffect(() => {
    setValueTags(map(value, String));
  }, [value]);

  const handleAddTag = tags => {
    const newTags = uniq(map(tags, tag => tag.trim()));

    setValueTags(newTags);
    onValueChange(newTags);
    setInputValue('');
  };

  const handleDeleteTag = tags => {
    setValueTags(tags);
    onValueChange(tags);
  };

  const handleInputChange = newInputValue => {
    setInputValue(newInputValue);
  };

  if (isReadOnly) {
    return (
      <span className={getPrefixedClasses(['--rule-value-readonly', '--text-space-left'])}>
        {valueTags.join(', ')}
      </span>
    );
  }

  return (
    <AddTags
      allowTagSplitting={false}
      id={id}
      inline
      inlineInputAlwaysOn
      inputValue={inputValue}
      name={id}
      onAddTag={handleAddTag}
      onDelete={handleDeleteTag}
      onInputChange={handleInputChange}
      placeholder={placeholder || t('Enter value')}
      tagExistsError={t('This value already exists')}
      tags={valueTags}
      validateInputValue
    />
  );
};

OneOfValues.propTypes = {
  id: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
  operator: PROPTYPES.OPERATOR,
  placeholder: PropTypes.string,
  value: PropTypes.arrayOf(PropTypes.string),
};

OneOfValues.defaultProps = {
  operator: null,
  placeholder: '',
  value: [],
};

export default OneOfValues;
