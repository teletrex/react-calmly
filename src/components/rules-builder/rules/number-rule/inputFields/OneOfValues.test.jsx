/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { IS_ONE_OF_OPERATORS } from '../../constants/operators';
import { renderWithContext } from '../../../utils/testHelpers';

import OneOfValues from './OneOfValues';

const initialValue = [7];

const defaultProps = {
  id: 'TestNumberRuleElement',
  onValueChange: jest.fn(),
};

describe('OneOfValues', () => {
  it('should render correctly', () => {
    const { container } = render(<OneOfValues {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with a provided value', () => {
    const { container } = render(<OneOfValues {...defaultProps} value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly for read-only', () => {
    const { container, getByText } = renderWithContext(
      <OneOfValues {...defaultProps} value={initialValue} />,
      { isReadOnly: true }
    );
    expect(container).toMatchSnapshot();
    expect(getByText(initialValue[0].toString())).toBeInTheDocument();
  });

  it('should set the rule value when numbers are provided', () => {
    const { getByPlaceholderText, getByText } = render(<OneOfValues {...defaultProps} />);
    const numberInput = getByPlaceholderText('Enter value');
    const addButton = getByText('Add');

    fireEvent.change(numberInput, { target: { value: initialValue[0] } });
    fireEvent.click(addButton);

    expect(defaultProps.onValueChange).toHaveBeenCalledWith(initialValue);
  });

  it('should only add the unique values', () => {
    const { getByPlaceholderText, getByText } = render(<OneOfValues {...defaultProps} />);
    const numberInput = getByPlaceholderText('Enter value');
    const addButton = getByText('Add');

    fireEvent.change(numberInput, { target: { value: initialValue[0] } });
    fireEvent.click(addButton);
    fireEvent.change(numberInput, { target: { value: initialValue[0] } });
    fireEvent.click(addButton);

    expect(defaultProps.onValueChange).toHaveBeenCalledWith(initialValue);
    expect(defaultProps.onValueChange).not.toHaveBeenCalledWith([initialValue[0], initialValue[0]]);
  });

  it('should correctly remove an existing value', () => {
    const { getByTitle } = render(<OneOfValues {...defaultProps} value={initialValue} />);
    const removeButton = getByTitle('Remove');

    fireEvent.click(removeButton);

    expect(removeButton).not.toBeInTheDocument();
  });

  it('should parse values for one of values component', () => {
    const initialValue = [3, 9];
    const { getByPlaceholderText, getByText } = render(
      <OneOfValues {...defaultProps} value={initialValue} />
    );

    fireEvent.change(getByPlaceholderText('Enter value'), { target: { value: '-0014' } });
    fireEvent.click(getByText('Add'));

    expect(defaultProps.onValueChange).toHaveBeenCalledWith([3, 9, -14]);

    fireEvent.change(getByPlaceholderText('Enter value'), { target: { value: '-00.014' } });
    fireEvent.click(getByText('Add'));

    expect(defaultProps.onValueChange).toHaveBeenCalledWith([3, 9, -14, -0.014]);
  });

  it('should reset validation on operator change', async () => {
    const props = {
      ...defaultProps,
      operator: IS_ONE_OF_OPERATORS[0],
      value: initialValue,
    };
    const { findByText, getByPlaceholderText, rerender } = render(<OneOfValues {...props} />);

    const numberInput = getByPlaceholderText('Enter value');
    fireEvent.change(numberInput, { target: { value: initialValue[0] } });
    fireEvent.blur(numberInput); // trigger validation
    const validationMessage = await findByText('This value already exists');

    expect(validationMessage).toBeInTheDocument();

    rerender(<OneOfValues {...props} operator={IS_ONE_OF_OPERATORS[1]} />);
    expect(validationMessage).not.toBeInTheDocument();
  });
});
