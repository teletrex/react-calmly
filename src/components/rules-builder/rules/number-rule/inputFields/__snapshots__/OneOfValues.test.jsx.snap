// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneOfValues should render correctly 1`] = `
<div>
  <div
    class="cds--add-tags cds--add-tags--inline cds--add-tags--inline-input-always-on"
  >
    <div
      class="cds--add-tags--container"
    >
      <div
        class="cds--add-tags--input-container"
      >
        <div
          class="cds--form-item cds--text-input-wrapper"
        >
          <span
            class="cds--text-input__label-wrapper cds--visually-hidden"
          >
            <label
              class="cds--label"
              for="TestNumberRuleElement"
            >
              Tags
            </label>

          </span>
          <div
            class="cds--text-input__field-wrapper"
          >
            <input
              aria-label="Tag input"
              class="cds--text-input cds--add-tags--input"
              id="TestNumberRuleElement"
              min="0"
              name="TestNumberRuleElement"
              placeholder="Enter value"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        aria-label="Add"
        class="cds--add-tags--submit cds--button__with-icon cds--btn cds--btn--sm cds--btn--ghost cds--btn--disabled cds--btn--icon-only cds--tooltip__trigger cds--tooltip--a11y cds--tooltip--bottom cds--tooltip--align-center"
        disabled=""
        tabindex="0"
        type="button"
      >
        <span
          class="cds--assistive-text"
        >
          Add
        </span>
        <svg
          aria-hidden="true"
          aria-label="Add"
          class="cds--btn__icon"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 10.6L2.5 7.1l-.7.7 3.5 3.5.7.7 7.1-7.1-.7-.7z"
          />
        </svg>
      </button>
      <button
        aria-label="Cancel"
        class="cds--add-tags--cancel cds--button__with-icon cds--btn cds--btn--sm cds--btn--ghost cds--btn--icon-only cds--tooltip__trigger cds--tooltip--a11y cds--tooltip--bottom cds--tooltip--align-center"
        tabindex="0"
        type="button"
      >
        <span
          class="cds--assistive-text"
        >
          Cancel
        </span>
        <svg
          aria-hidden="true"
          aria-label="Cancel"
          class="cds--btn__icon"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`OneOfValues should render correctly for read-only 1`] = `
<div>
  <span
    class="cds--rule-value-readonly cds--text-space-left"
  >
    7
  </span>
</div>
`;

exports[`OneOfValues should render correctly with a provided value 1`] = `
<div>
  <div
    class="cds--add-tags cds--add-tags--inline cds--add-tags--inline-input-always-on"
  >
    <div
      class="cds--tag cds--tag--gray cds--add-tags--filter-tag cds--tag--filter cds--tag--oval"
      id="tag-7"
      title=""
    >
      <div
        class="cds--tag__label"
      >
        <span
          class="cds--tag__text-ellipsis"
        >
          7
        </span>
      </div>
      <button
        aria-label="Remove"
        title="Remove"
        type="button"
      >
        <svg
          aria-hidden="true"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          style="will-change: transform;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"
          />
        </svg>
      </button>
    </div>
    <div
      class="cds--add-tags--container"
    >
      <div
        class="cds--add-tags--input-container"
      >
        <div
          class="cds--form-item cds--text-input-wrapper"
        >
          <span
            class="cds--text-input__label-wrapper cds--visually-hidden"
          >
            <label
              class="cds--label"
              for="TestNumberRuleElement"
            >
              Tags
            </label>

          </span>
          <div
            class="cds--text-input__field-wrapper"
          >
            <input
              aria-label="Tag input"
              class="cds--text-input cds--add-tags--input"
              id="TestNumberRuleElement"
              min="0"
              name="TestNumberRuleElement"
              placeholder="Enter value"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        aria-label="Add"
        class="cds--add-tags--submit cds--button__with-icon cds--btn cds--btn--sm cds--btn--ghost cds--btn--disabled cds--btn--icon-only cds--tooltip__trigger cds--tooltip--a11y cds--tooltip--bottom cds--tooltip--align-center"
        disabled=""
        tabindex="0"
        type="button"
      >
        <span
          class="cds--assistive-text"
        >
          Add
        </span>
        <svg
          aria-hidden="true"
          aria-label="Add"
          class="cds--btn__icon"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 10.6L2.5 7.1l-.7.7 3.5 3.5.7.7 7.1-7.1-.7-.7z"
          />
        </svg>
      </button>
      <button
        aria-label="Cancel"
        class="cds--add-tags--cancel cds--button__with-icon cds--btn cds--btn--sm cds--btn--ghost cds--btn--icon-only cds--tooltip__trigger cds--tooltip--a11y cds--tooltip--bottom cds--tooltip--align-center"
        tabindex="0"
        type="button"
      >
        <span
          class="cds--assistive-text"
        >
          Cancel
        </span>
        <svg
          aria-hidden="true"
          aria-label="Cancel"
          class="cds--btn__icon"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
